src/gluonts/distribution/bijection.py

```diff
@@ -229,8 +229,8 @@ def f_inv(self, y: Tensor) -> Tensor:
     def log_abs_det_jac(self, x: Tensor, y: Tensor) -> Tensor:
         if self.scale is not None:
             return self.scale.broadcast_like(x).abs().log()
-        else:
-            raise RuntimeError("scale is of type None in log_abs_det_jac")
+        F = getF(x)
+        return F.zeros_like(x)
```

Comment:
I think we should avoid silently doing something with undefined inputs. At the very least, I would issue a warning here.

Reply:
The reason for this change is that when `scale is None`, the `f` and `f_inv` methods act like if `scale == 1`: in this case the Jacobian is exactly `scale`, and if `scale == 1` then `log(abs(det(scale))) == 0`, hence the returned value.

To summarize, in my opinion `None` here is not an undefined input, but rather a way of encoding the default `1` (for `scale`, and `0` for `loc`). In fact, no warning is issued in `f` or `f_inv` when either `scale` or `loc` are `None`.

`F.zeros_like(x)` is exactly log|det(J)| in case `scale is None`, where the Jacobian J in this case is precisely `scale`: if you check what the implementations of `f` and `f_inv`, they act like if `scale = 1`, in which case `log(abs(det(scale))) == 0`

Reply:
I see. Wouldn't it then be cleaner to just set the defaults in the constructor when `None` is passed and avoid the `if sth is None` logic throughout the class?

Reply:
> I see. Wouldn't it then be cleaner to just set the defaults in the constructor when `None` is passed and avoid the `if sth is None` logic throughout the class?

Hmm, maybe not. It's tricky because we don't know the shape of the input tensor if the values are `None`.

Reply:
> I see. Wouldn't it then be cleaner to just set the defaults in the constructor when `None` is passed and avoid the `if sth is None` logic throughout the class?

Hmm, maybe not. It's tricky because we don't know the shape of the input tensor if the values are `None`.

Reply:
Yes, exactly. I think having `None` as default is not too bad: when `loc` (resp. `scale`) is `None`, it is the additive (multiplicative) identity. It just needs to be clear: maybe I'll specify this better in the docstring.